components/org.wso2.carbon.identity.context.rewrite.valve/src/main/java/org/wso2/carbon/identity/context/rewrite/valve/OrganizationContextRewriteValve.java

```diff
@@ -141,45 +151,50 @@ private void unsetMDCThreadLocals() {
-                                new QName(IdentityCoreConstants.IDENTITY_DEFAULT_NAMESPACE, "BasePath"));
-                        if (basePath != null) {
-                            OMElement subPaths = context.getFirstChildWithName(
-                                    new QName(IdentityCoreConstants.IDENTITY_DEFAULT_NAMESPACE, "SubPaths"));
-                            List<String> subPathList = null;
-                            if (subPaths != null) {
-                                Iterator paths = subPaths.getChildrenWithName(
-                                        new QName(IdentityCoreConstants.IDENTITY_DEFAULT_NAMESPACE, "Path"));
-                                if (paths != null) {
-                                    subPathList = new ArrayList<>();
-                                    while (paths.hasNext()) {
-                                        OMElement subPath = (OMElement) paths.next();
-                                        subPathList.add(subPath.getText());
-                                    }
-                                }
-                            }
-                            // Honor the config order.
-                            if (!rewriteContexts.containsKey(basePath.getText())) {
-                                rewriteContexts.put(basePath.getText(), subPathList);
-                            }
-                        }
-                    }
+    private List<OrganizationRewriteContext> getOrgContextsToRewrite() {
+
+        List<OrganizationRewriteContext> organizationRewriteContexts = new ArrayList<>();
+        Map<String, Object> configuration = IdentityConfigParser.getInstance().getConfiguration();
+        Object webAppBasePathContexts = configuration.get("OrgContextsToRewrite.WebApp.Context.BasePath");
+        setOrganizationRewriteContexts(organizationRewriteContexts, webAppBasePathContexts, true);
+
+        Object webAppSubPathContexts = configuration.get("OrgContextsToRewrite.WebApp.Context.SubPaths.Path");
+        setSubPathContexts(organizationRewriteContexts, webAppSubPathContexts);
+
+        Object servletBasePathContexts = configuration.get("OrgContextsToRewrite.Servlet.Context");
+        setOrganizationRewriteContexts(organizationRewriteContexts, servletBasePathContexts, false);
+
+        return organizationRewriteContexts;
+    }
+
+    private void setOrganizationRewriteContexts(List<OrganizationRewriteContext> organizationRewriteContexts,
+                                                Object basePathContexts, boolean isWebApp) {
+
+        if (basePathContexts != null) {
+            if (basePathContexts instanceof ArrayList) {
+                for (String context : (ArrayList<String>) basePathContexts) {
+                    organizationRewriteContexts.add(new OrganizationRewriteContext(isWebApp, context));
+                }
+            } else {
+                organizationRewriteContexts.add(new OrganizationRewriteContext(true,
```

Comment:
Any reason to hardcode "true" here, instead of using the isWebApp param value?

Reply:
It was a mistake. Fixed by - https://github.com/wso2-extensions/identity-carbon-auth-rest/pull/205/commits/1b1e7c32c7f8ac267866f9fdd812932995510312